persistent-test/src/MigrationTest.hs

```diff
@@ -30,21 +31,43 @@ Source1 sql=source
     field3 Int
     extra  Int
     field4 Target1Id
+
 |]
 
-specsWith :: (MonadIO m) => RunDb SqlBackend m -> Spec
+share [mkPersist sqlSettings, mkMigrate "addPrimKey", mkDeleteCascade sqlSettings] [persistLowerCase|
+FromRawMigration
+    name T.Text
+    age  Int
+
+    Primary name
+
+|]
+
+specsWith :: (MonadIO m, MonadFail m) => RunDb SqlBackend m -> Spec
 specsWith runDb = describe "Migration" $ do
     it "is idempotent" $ runDb $ do
-      again <- getMigration migrationMigrate
-      liftIO $ again @?= []
-    it "really is idempotent" $ runDb $ do
-      runMigration migrationMigrate
-      again <- getMigration migrationMigrate
-      liftIO $ again @?= []
+        _ <- runMigration migrateionMigrate
+        again <- getMigration migrationMigrate
+        liftIO $ again @?= []
     it "can add an extra column" $ runDb $ do
-      -- Failing test case for #735.  Foreign-key checking, switched on in
-      -- version 2.6.1, caused persistent-sqlite to generate a `references`
-      -- constraint in a *temporary* table during migration, which fails.
-      _ <- runMigration migrationAddCol
-      again <- getMigration migrationAddCol
-      liftIO $ again @?= []
+        -- Failing test case for #735.  Foreign-key checking, switched on in
+        -- version 2.6.1, caused persistent-sqlite to generate a `references`
+        -- constraint in a *temporary* table during migration, which fails.
+        _ <- runMigration migrationMigrate
+        _ <- runMigration migrationAddCol
+        again <- getMigration migrationAddCol
+        liftIO $ again @?= []
+    describe "Add Primary key constraint on raw table" $ do
+        it "should not be considered safe" $ runDb $ do
```

Comment:
Adding a primary key to a column is definitely *not* safe, because it's like adding a unique key to a column.